### PR TITLE
Legacyp12 playbook

### PR DESCRIPTION
**Pull request by Pmaraveyias**
Addresses Issue #431. Allows the use of legacy p12 as an option in playbooks, but must be specifically chosen.

### CODE REVIEW

File: pkg/playbook/app/installer/capi.go [hunk 1]
```diff
@@ -112,7 +112,7 @@ func (r CAPIInstaller) Install(pcc certificate.PEMCollection) error {
 	// Generate random password for temporary P12 bundle
 	bundlePassword := vcertutil.GeneratePassword()
 
-	content, err := packageAsPKCS12(pcc, bundlePassword)
+	content, err := packageAsPKCS12(pcc, bundlePassword, false)
```
Comment (BeardedPrincess) [inline pkg/playbook/app/installer/capi.go:115]: wouldn't this need to be `packageAsPKCS12(pcc, bundlePassword, r.UseLegacyP12)` as well?

Comment (Pmaraveyias) [reply pkg/playbook/app/installer/capi.go]: Yep, fixed

**Comment by BeardedPrincess** [comment]
I added a PR to update the playbook readme as well: #465

**Comment by BeardedPrincess** [comment]
Validated using Windows Server 2016 and the useLegacyPkcs12 flag in CAPI installation playbook worked.  Also verified it's still broken if that flag is absent or set to `false`.

Additionally, tested both toggles `true` and `false` with an installation type of PKCS12.  Worked as expected, based on openssl results: 

`useLegacyPkcs12: true` :
```
MAC: sha1, Iteration 1
MAC length: 20, salt length: 8
PKCS7 Encrypted data: pbeWithSHA1And3-KeyTripleDES-CBC, Iteration 2048
Certificate bag
Certificate bag
Certificate bag
PKCS7 Data
Shrouded Keybag: pbeWithSHA1And3-KeyTripleDES-CBC, Iteration 2048
```

File: pkg/playbook/app/domain/installation.go [hunk 2]
```diff
@@ -51,9 +51,10 @@ type Installation struct {
 	KeyFile             string `yaml:"keyFile,omitempty"`
 	KeyPassword         string `yaml:"keyPassword,omitempty"`
 	// Deprecated: Location is deprecated in favor of CAPILocation. It will be removed on a future release
-	Location    string             `yaml:"location,omitempty"`
-	P12Password string             `yaml:"p12Password,omitempty"`
-	Type        InstallationFormat `yaml:"format,omitempty"`
+	Location     string             `yaml:"location,omitempty"`
+	P12Password  string             `yaml:"p12Password,omitempty"`
+	UseLegacyP12 bool               `yaml:"useLegacyPkcs12,omitempty"`
```
Comment (rvelaVenafi) [inline pkg/playbook/app/domain/installation.go:56]: can we change the yaml label to `useLegacyP12` to be consistent with the other P12 flag `p12Password`.

**Review by rvelaVenafi — CHANGES_REQUESTED** [review]

**Review by marcos-albornoz — APPROVED** [review]

**Review by luispresuelVenafi — APPROVED** [review]

File: pkg/playbook/app/domain/installation.go (hunk 2 above)
Comment (luispresuelVenafi) [reply pkg/playbook/app/domain/installation.go]: this change is done by latest commit

**Review by rvelaVenafi — APPROVED** [review]

**Pull request merged**